### PR TITLE
um7: depend on oasis3-mct branch access-esm1.5

### PR DESCRIPTION
**Pull request by harshula**

### CODE REVIEW

**Review by penguian — CHANGES_REQUESTED** [review]
The comparison https://github.com/ACCESS-NRI/oasis3-mct/compare/access-esm1.5...ACCESS-NRI:oasis3-mct:access-esm1.5-new-modules shows non-trivial differences between the two branches. Are each of these differences intended, and what are their purpose?

**Comment by penguian** [comment]
Additionally, my first attempts at installation failed:
```
[pcl851@gadi-login-09 spack-packages]$ spack install um7%intel
==> Error: concretization failed for the following reasons:

   1. Cannot satisfy 'oasis3-mct@git.access-esm1.5'
   2. Cannot satisfy 'oasis3-mct@git.access-esm1.5'
        required because um7 depends on oasis3-mct@git.access-esm1.5 
          required because um7%intel requested explicitly 
[pcl851@gadi-login-09 spack-packages]$ spack install oasis3-mct@git.access-esm1.5%intel
==> Error: access-esm1.5 is not a valid git ref for oasis3-mct
```

**Comment by harshula** [comment]
Hi @penguian , I've updated the `access-esm1.5` branch. Can you please verify again?

**Comment by harshula** [comment]
> spack install oasis3-mct@git.access-esm1.5%intel
> ==> Error: access-esm1.5 is not a valid git ref for oasis3-mct

```
# spack install oasis3-mct@git.access-esm1.5 %intel@2021.2.0 arch=linux-rocky8-x86_64
...

**Comment by harshula** [comment]
> spack install oasis3-mct@git.access-esm1.5%intel
> ==> Error: access-esm1.5 is not a valid git ref for oasis3-mct

```
# spack install oasis3-mct@git.access-esm1.5 %intel@2021.2.0 arch=linux-rocky8-x86_64
...
==> Installing oasis3-mct-git.access-esm1.5=0-git.71-vuzjsdgqpjwgb55nb2gwhiafaqtqtfl5
==> No binary for oasis3-mct-git.access-esm1.5=0-git.71-vuzjsdgqpjwgb55nb2gwhiafaqtqtfl5 found: installing from source
==> No patches needed for oasis3-mct
==> oasis3-mct: Executing phase: 'edit'
==> oasis3-mct: Executing phase: 'build'
==> oasis3-mct: Executing phase: 'install'
==> oasis3-mct: Successfully installed oasis3-mct-git.access-esm1.5=0-git.71-vuzjsdgqpjwgb55nb2gwhiafaqtqtfl5
  Stage: 2.55s.  Edit: 0.00s.  Build: 2m 4.96s.  Install: 0.03s.  Post-install: 0.19s.  Total: 2m 7.89s
[+] /opt/release/linux-rocky8-x86_64/intel-2021.2.0/oasis3-mct-git.access-esm1.5=0-git.71-vuzjsdgqpjwgb55nb2gwhiafaqtqtfl5
```

File: packages/oasis3-mct/package.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Oasis3Mct(MakefilePackage):
     maintainers = ["harshula"]
 
     version("master", branch="master")
+    version("access-esm1.5", branch="access-esm1.5")
```
Comment (aidanheerdegen) [inline packages/oasis3-mct/package.py:21]: I was going to ask if the `access-esm1.5` branch is protected, but I see none of them are. Have made an issue

https://github.com/ACCESS-NRI/oasis3-mct/issues/22

This is a comment, it does not require any changes.

**Review by aidanheerdegen — CHANGES_REQUESTED** [review]
The Ruff checks should be addressed. If we're not using `pprint` then we shouldn't import it. The f-string errors are just pointing out that there are no variable interpolations in those strings, so they can just be simple strings with the `f`.

**Comment by harshula** [comment]
Hi @aidanheerdegen , Will address linter issues when adding support for oneapi compilers.

**Review by aidanheerdegen — APPROVED** [review]
Ok to leave linter changes for now.

**Pull request merged**